### PR TITLE
Feature/student details

### PR DESCRIPTION
**Pull request by KyleWong2510**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- No student details displayed
## What is the new behavior?

- Add functionality for displaying student details
- GET method for scores
- GET method for average scores
- Fix propTypes errors in tests
- Pass thunks tests for getStudentsResults and getStudentAverage

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/CreateLesson/CreateLesson.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 
-const CreateLesson = () => {
+const CreateLesson = ({ resetStudentsResults }) => {
   const [questions, setQuestions] = useState([])
   const [lessonTitleText, setLessonTitleText] = useState("");
   
+  useEffect (() => {
+    resetStudentsResults()
+  })
+
```
Comment (Melizzo) [inline src/CreateLesson/CreateLesson.js:16]: Great thinking here Kyle - I think I actually dreamt a similar solution last night!

File: src/CreateStudentForm/CreateStudentForm.test.js [hunk 2]
```diff
@@ -9,14 +9,22 @@ import { rootReducer } from '../reducers';
...
 })
+let createStudentForm
+let mockCompleteForm
 
 describe('CreateStudent', () => {
-  it('should render a header, two inputs, and a submit btn', () => {
-    const { getByText, getByPlaceholderText, getByDisplayValue } = render(
+  beforeEach(() => {
+
+    mockCompleteForm = jest.fn()
```
Comment (Melizzo) [inline src/CreateStudentForm/CreateStudentForm.test.js:18]: Love the beforeEach

File: src/Modal/Modal.test.js [hunk 3]
```diff
@@ -3,10 +3,23 @@ import '@testing-library/jest-dom/extend-expect';
...
+        content={mockContent}
+        toggleDisplay={mockToggleDisplay}
+      />
+    )
+  })
+
   it('should render a button and some content', () => {
     const mockContent = <p>Content</p>
-    const { getByText, getByRole } = render(<Modal content={mockContent}/>)
+    const { getByText, getByRole } = render(modal)
```
Comment (Melizzo) [inline src/Modal/Modal.test.js:23]: smart thinking of the beforeEach with the mocked modal Kyle.

**Review by Melizzo — APPROVED** [review]

File: src/QuestionFormCard/AnswerInput/AnswerInput.test.js [hunk 4]
```diff
@@ -3,22 +3,48 @@ import '@testing-library/jest-dom/extend-expect';
 import { render, fireEvent } from '@testing-library/react';
 import AnswerInput from './AnswerInput';
 
+let mockSetAnswerText
+
 describe('AnswerInput', () => {
+  beforeEach(() => {
+    mockSetAnswerText = jest.fn()
+  })
   it('should render an input with a message if the answer is the correct answer', () => {
```
Comment (Billwilke42) [inline src/QuestionFormCard/AnswerInput/AnswerInput.test.js:12]: nice work on these tests

File: src/StudentDetails/StudentDetails.js [hunk 5]
```diff
@@ -1,12 +1,80 @@
...
+    lessons.forEach(lesson => {
+      if(lesson.id === result.lesson) {
+        acc.push({
+          lesson: lesson.name,
+          score: result.score,
+          response: result.mood,
+          mood: result.mood_analyzer
+        })
+      }
+    })
```
Comment (Billwilke42) [inline src/StudentDetails/StudentDetails.js:41]: we should change the acc to be more semantic

File: src/TeacherDashboard/TeacherDashboard.js [hunk 6]
```diff
@@ -65,5 +68,7 @@ export default connect(mapStateToProps, mapDispatchToProps)(TeacherDashboard);
 TeacherDashboard.propTypes = {
   getTeacher: PropTypes.func.isRequired,
   getStudents: PropTypes.func.isRequired,
-  getLessons: PropTypes.func.isRequired
+  getLessons: PropTypes.func.isRequired,
+  resetStudentsResults: PropTypes.func.isRequired,
```
Comment (Billwilke42) [inline src/TeacherDashboard/TeacherDashboard.js:72]: some of these isRequired are throwing errors on my side

**Review by Billwilke42 — APPROVED** [review]
Nice work! Almost done

**Pull request merged**